src/dbnode/storage/index.go

```diff
@@ -1334,6 +1353,52 @@ func (i *nsIndex) Query(
 	}, nil
 }
 
+func (i *nsIndex) WideQuery(
+	ctx context.Context,
+	query index.Query,
+	opts index.WideQueryOptions,
+) error {
+	logFields := []opentracinglog.Field{
+		opentracinglog.String("wideQuery", query.String()),
+		opentracinglog.String("namespace", i.nsMetadata.ID().String()),
+		opentracinglog.Int("batchSize", opts.BatchSize),
+		xopentracing.Time("queryStart", opts.StartInclusive),
+		xopentracing.Time("queryEnd", opts.EndExclusive),
+	}
+
+	ctx, sp := ctx.StartTraceSpan(tracepoint.NSIdxWideQuery)
+	sp.LogFields(logFields...)
+	defer sp.Finish()
+
+	results := index.NewWideQueryResults(
+		i.nsMetadata.ID(),
+		i.opts.IdentifierPool(),
+		i.shardForID(),
+		opts,
+	)
+
+	ctx.RegisterFinalizer(results)
+	queryOpts := opts.ToQueryOptions()
+
+	closed := false
+	defer func() {
+		if !closed {
+			// Drain any remaining results and close.
+			results.Finalize()
+		}
+	}()
+
+	_, err := i.query(ctx, query, results, queryOpts, i.execBlockWideQueryFn, logFields)
+	if err != nil {
```

Comment:
nit: can join to previous line

Reply:
Usually try and avoid it for longer lines, and this is already over 80 :p